src/test/java/io/stargate/sgv2/jsonapi/util/YamlMergerTest.java

```diff
@@ -0,0 +1,416 @@
+package io.stargate.sgv2.jsonapi.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class YamlMergerTest {
```

Comment:
we should simplify these tests to just be about testing the merging without the complexity of the embedding config or other faked examples or reading other config files. 

so very simple tests that make it easy to see the yaml changes. 

will need to load the yaml into a JsonNode in this test class because of other changes on the YamlMerger - see comments there.